livestream-compose-sample/src/main/java/io/getstream/livestream/compose/ui/LiveStreamHomeActivity.kt

```diff
@@ -23,98 +26,134 @@ import androidx.compose.ui.Modifier
 import io.getstream.livestream.compose.R
 import io.getstream.livestream.compose.darkColorPalette
 import io.getstream.livestream.compose.lightColorPalette
+import io.getstream.livestream.compose.shapes
 
 class LiveStreamHomeActivity : ComponentActivity() {
-    private val factory by lazy {
-        ChannelViewModelFactory(
-            ChatClient.instance(), ChatDomain.instance(),
-            QuerySort.desc("last_updated"),
-            Filters.and(
-                Filters.eq("type", "livestream"),
-                Filters.`in`("members", listOf(ChatClient.instance().getCurrentUser()?.id ?: ""))
-            )
-        )
-    }
-
-    private val listViewModel: ChannelListViewModel by viewModels { factory }
+    private lateinit var sharedPref: SharedPreferences
 
     @ExperimentalMaterialApi
     @ExperimentalFoundationApi
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        sharedPref =
+            getSharedPreferences(getString(R.string.name_shared_pref), Context.MODE_PRIVATE)
 
         setContent {
             var expanded by remember { mutableStateOf(false) }
-            var isDarkMode by remember { mutableStateOf(false) }
+            var isDarkMode by remember {
+                mutableStateOf(
+                    sharedPref.getBoolean(
+                        getString(R.string.key_theme),
+                        false
+                    )
+                )
+            }
             var isGrid by remember { mutableStateOf(true) }
 
-            ChatTheme(colors = if (isDarkMode) darkColorPalette() else lightColorPalette()) {
-                CustomHeader(
-                    title = resources.getString(R.string.app_name),
-                    actions = {
-                        //Adds a icon button for switching theme
-                        IconButton(
-                            onClick = {
-                                expanded = !expanded
-                            }) {
-                            Image(
-                                painter = painterResource(id = R.drawable.ic_theme_switch),
-                                contentDescription = stringResource(id = R.string.accessibilitySwitchTheme),
-                                modifier = Modifier
-                                    .width(24.dp)
-                                    .height(24.dp)
-                            )
-                        }
-                        //Theme chooser drop down selector
-                        DropdownMenu(
-                            expanded = expanded,
-                            onDismissRequest = { expanded = false }
-                        ) {
+            ChatTheme(
+                colors = if (isDarkMode) darkColorPalette() else lightColorPalette(),
+                shapes = shapes()
+            ) {
+                Scaffold(
+                    topBar = {
+                        TopAppBar(
```

Comment:
Can we split these components up a little? Something like a `LiveStreamHomeTopBar`.